producers/prometheus/prometheus_test.go

```diff
@@ -37,58 +41,185 @@ func TestNew(t *testing.T) {
+								&dto.LabelPair{Name: proto.String("hostname"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("mesos_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("tag1"), Value: proto.String("value1")},
+								&dto.LabelPair{Name: proto.String("tag2"), Value: proto.String("")}, // empty
+								&dto.LabelPair{Name: proto.String("task_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("task_name"), Value: proto.String("")},
+							},
+						},
+						&dto.Metric{
+							Gauge: &dto.Gauge{Value: proto.Float64(789.012)},
+							Label: []*dto.LabelPair{
+								&dto.LabelPair{Name: proto.String("cluster_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("container_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("executor_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_name"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_principal"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("framework_role"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("hostname"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("mesos_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("tag1"), Value: proto.String("")}, // empty
+								&dto.LabelPair{Name: proto.String("tag2"), Value: proto.String("value2")},
+								&dto.LabelPair{Name: proto.String("task_id"), Value: proto.String("")},
+								&dto.LabelPair{Name: proto.String("task_name"), Value: proto.String("")},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
 
-		// The web server should be listening on the configured TCP port
-		_, err = net.Dial("tcp", net.JoinHostPort("localhost", strconv.Itoa(port)))
-		So(err, ShouldBeNil)
+	// Ensure the environment is clean
+	if err := os.Unsetenv("LISTEN_PID"); err != nil {
+		panic(err)
+	}
+	if err := os.Unsetenv("LISTEN_FDS"); err != nil {
+		panic(err)
+	}
 
-		// feed in some fake data
-		dps := []producers.Datapoint{
-			producers.Datapoint{
-				Name:  "datapoint-one",
-				Value: 123.456,
-			},
-		}
-		metricsChan <- producers.MetricsMessage{
-			Name:       "some-message",
-			Datapoints: dps,
-			Timestamp:  time.Now().UTC().Unix(),
-		}
-		time.Sleep(250 * time.Millisecond)
+	for _, testCase := range testCases {
+		Convey("Should create a Prometheus server and emit metrics: "+testCase.name, t, func() {
+			// Find an available TCP port.
+			port, err := getEphemeralPort()
+			if err != nil {
+				panic(err)
+			}
 
-		resp, err := http.Get("http://" + net.JoinHostPort("localhost", strconv.Itoa(port)) + "/metrics")
-		So(err, ShouldBeNil)
-		defer resp.Body.Close()
+			// Start the server.
+			p, metricsChan := New(Config{Port: port})
+			go p.Run()
+			time.Sleep(1 * time.Second)
```

Comment:
Do we really need to sleep here? What are we waiting for? Please replace it with a channel (if possible) or reduce this to a millisecond.

Reply:
This is how the test already behaves. It's not relevant to the bug I'm trying to fix with this PR, and this repo is deprecated anyway, so I'd like to avoid diving down that rabbit hole unless it presents a practical problem.